src/index.ts

```diff
@@ -21,33 +22,59 @@ export class TestxServer {
   private schema: string;
   private server: Server;
   private serverUrl: string;
+  private expressServer: Express;
   private dbConnection: Knex;
 
   constructor(schema: string) {
     this.schema = schema;
   }
 
   public async start() {
-    const { server, url } = await this.generateServer();
+    const port = await getAvailablePort();
```

Comment:
We can only solve this problem by using some library to control ports. Because even storing the port as a property, the port still is available to be selected by getAvailablePort function.

Reply:
What we could do is exposing the server to random port in a bigger port range, so that the probability that two servers pick the same port will be really low, almost none, and a second alternative could be to retry until the port doesn't become available (using a timeout).
I think the first one should be enough, wdyt?

Reply:
Yes, I agree

Reply:
I've opened a separate issue for this problem: https://github.com/aerogear/graphql-testx/issues/57